tests/pipeline/test_dispatcher.py

```diff
@@ -0,0 +1,140 @@
+from zipline.pipeline import USEquityPricingLoader
+from zipline.pipeline.data import (
+    Column,
+    DataSet,
+    BoundColumn,
+    USEquityPricing,
+)
+from zipline.pipeline.dispatcher import (
+    PipelineDispatcher,
+    clear_all_associations,
+)
+from zipline.pipeline.loaders.base import PipelineLoader
+from zipline.pipeline.sentinels import NotSpecified
+from zipline.testing import ZiplineTestCase
+from zipline.testing.fixtures import WithAdjustmentReader
+from zipline.testing.predicates import (
+    assert_raises_str,
+    assert_equal,
+)
+from zipline.utils.numpy_utils import float64_dtype
+
+
+class FakeDataSet(DataSet):
+    test_col = Column(float64_dtype)
+
+
+class FakeColumn(BoundColumn):
+    pass
+
+
+class FakePipelineLoader(PipelineLoader):
+
+    def load_adjusted_array(self, columns, dates, assets, mask):
+        pass
+
+
+class UnrelatedType(object):
+    pass
+
+
+class PipelineDispatcherTestCase(WithAdjustmentReader, ZiplineTestCase):
+
+    @classmethod
+    def init_class_fixtures(cls):
+        super(PipelineDispatcherTestCase, cls).init_class_fixtures()
+        cls.default_pipeline_loader = USEquityPricingLoader(
+            cls.bcolz_equity_daily_bar_reader,
+            cls.adjustment_reader,
+        )
+
+        cls.add_class_callback(clear_all_associations)
+
+    def test_load_not_registered(self):
+        fake_col_instance = FakeColumn(
```

Comment:
I haven't looked carefully at these tests yet, but in general, it's not expected that anyone should ever construct BoundColumn instances explicitly. The usual way to get a bound column is to do:

```
class SomeDataSet(DataSet):
    column = Column(...)

SomeDataSet.column  # This evaluates to a BoundColumn because Column implements the descriptor protocol.
```